pkg/splunk/enterprise/util.go

```diff
@@ -1895,3 +1895,98 @@ func handleAppFrameworkActivity(client splcommon.ControllerClient, cr splcommon.
 
 	return finalResult
 }
+
+// checkForAppFrameworkMigration upgrades the app framework status context
+func checkForAppFrameworkMigration(client splcommon.ControllerClient, cr splcommon.MetaObject, afwStatusContext *enterpriseApi.AppDeploymentContext, afwConf *enterpriseApi.AppFrameworkSpec, isLocalScope bool) error {
+	// If needed, Migrate the app framework status
+	if isAppFrameworkMigrationNeeded(afwStatusContext) {
+		// Spec validation updates the status with some of the defaults, which may not be there in older app framework versions
+		err := ValidateAppFrameworkSpec(afwConf, afwStatusContext, isLocalScope)
+		if err != nil {
+			return err
+		}
+
+		if !migrateAfwStatus(client, cr, afwStatusContext) {
+			return fmt.Errorf("app framework migration failed")
+		}
+
+		// Update the new status
+		err = client.Status().Update(context.TODO(), cr)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// migrateAfwStatus (if required)migrates the app frame work status context to the latest version
+func migrateAfwStatus(client splcommon.ControllerClient, cr splcommon.MetaObject, afwStatusContext *enterpriseApi.AppDeploymentContext) bool {
+	scopedLog := log.WithName("migrateAfwStatus").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	// This should never happen
+	if afwStatusContext == nil {
```

Comment:
I think this check is duplicate here since you have already checked it in `isAppFrameworkMigrationNeeded`